File: LAB/LAB2/Java/Solution.java
class Solution {
    public boolean isAnagram(String s, String t) {
        int[] count = new int[26];

        // Count the frequency of characters in string s
        for (char i : s.toCharArray()) {
            count[i - 'a']++;
        }

        // Decrement the frequency of characters in string t
        for (char i : t.toCharArray()) {
            count[i - 'a']--;
        }

        // Check if any character has non-zero frequency
        for (int c : count) {
            if (c != 0) {
                return false;
            }
        }

        return true;
    }
}

File: LAB/LAB2/Python/Solution.py
''' Brute Force
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        sorted_s = sorted(s)
        sorted_t = sorted(t)
        return sorted_s == sorted_t
'''
# Hash Table
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        count = [0] * 26
        
        # Count the frequency of characters in string s
        for x in s:
            count[ord(x) - ord('a')] += 1
        
        # Decrement the frequency of characters in string t
        for x in t:
            count[ord(x) - ord('a')] -= 1
        
        # Check if any character has non-zero frequency
        for c in count:
            if c != 0:
                return False
        
        return True